# Results comparison for MIST with fine-tuned models of model A partial

In this notebook, I will compare the results of the different fine tuned models (only the MIST dataset).

In [1]:
from ML.utils.utils import *
from physics.Iso_data_handler import Iso_data_handler
from ML.utils.Data_preparator import Data_preparator

In [2]:
pre_path = "../../../../../../"
path_default = pre_path + "results/model_A_partial/default_models"
path_fine_tuned = pre_path + "results/model_A_partial/fine_tuned_models"

## Getting the ranges for the values

In [3]:
iso_handler = Iso_data_handler("C:/Users/antoi/Code/unif/MA2/thesis/data/MIST_v1.2_vvcrit0.0_basic_isos/", 
                                    ['log10_isochrone_age_yr', 'log_Teff', 'log_g', 'phase', 'metallicity', 'star_mass', 'log_R', 'log_L'], 
                                    "MIST", reclassify=True)

iso_df = iso_handler.get_isochrone_dataframe()
display(iso_df)

Reading MIST dataframe from MIST_reclassified_iso_full_data.csv file...


,log10_isochrone_age_yr,log_Teff,log_g,phase,metallicity,star_mass,log_R,log_L
0,5.0,3.486221,3.131342,-1.0,-0.25,0.100000,0.153402,-0.793667
1,5.0,3.487362,3.126808,-1.0,-0.25,0.102645,0.160326,-0.775254
2,5.0,3.489243,3.119367,-1.0,-0.25,0.107039,0.171785,-0.744810
3,5.0,3.491102,3.112165,-1.0,-0.25,0.111419,0.183099,-0.714749
4,5.0,3.492937,3.105143,-1.0,-0.25,0.115789,0.194305,-0.684996
...,...,...,...,...,...,...,...,...
1467117,10.3,4.402490,7.777159,6.0,0.50,0.532726,-1.806255,-1.047903
1467118,10.3,4.387132,7.783242,6.0,0.50,0.532730,-1.809295,-1.115415
1467119,10.3,4.371789,7.789130,6.0,0.50,0.532735,-1.812237,-1.182672
1467120,10.3,4.356480,7.794844,6.0,0.50,0.532741,-1.815091,-1.249617


In [4]:
phase_filtered_iso_df = Data_preparator.filter_data(iso_df, {'phase':[0, 2, 3], 'star_mass':("<", 30)})
print_all_uniques(phase_filtered_iso_df)

log10_isochrone_age_yr  : Range : 5.0 - 10.3, Mean : 8.2913, Median : 8.3

log_Teff  : Range : 3.34 - 5.3542, Mean : 3.9814, Median : 3.9163

log_g  : Range : -1.08 - 6.2434, Mean : 3.5356, Median : 3.9078

Values in phase column : 0.0, 2.0, 3.0 

Values in metallicity column : -0.25, -0.5, -0.75, -1.0, -1.25, -1.5, -1.75, -2.0, -2.5, -3.0, -3.5, -4.0, 0.0, 0.25, 0.5 

star_mass  : Range : 0.1 - 29.9991, Mean : 5.0421, Median : 2.422

log_R  : Range : -1.0 - 3.1298, Mean : 0.6593, Median : 0.6146

log_L  : Range : -3.33 - 5.9763, Mean : 2.1988, Median : 2.3277



## Default vs fine-tuned

### KNN

In [5]:
compare_metrics(path_default, value_rounding=3, relative_values=False,
                output_parameters=["mass", "radius"],
                physical_models=["MIST"],
                model_names=["KNN"],
                data_filters=["scale_standard_14"],
                categories=["Global"])

Results for the Global category with a value of 1.0


value range  RVE  \
model physical_model filter            output_parameter                        
KNN   MIST           scale_standard_14 mass                0.1 - 13.999  1.0   
                                       radius            -0.997 - 3.082  1.0   

                                                          RMSE    MAE  MedAE  \
model physical_model filter            output_parameter                        
KNN   MIST           scale_standard_14 mass              0.057  0.022  0.005   
                                       radius            0.006  0.004  0.002   

                                                         CORR  MAX_ER  \
model physical_model filter            output_parameter                 
KNN   MIST           scale_standard_14 mass               1.0   2.613   
                                       radius             1.0   0.127   

                                                                                                         Percentiles  \
model physical_model filter            output_parameter                                                                
KNN   MIST           scale_standard_14 mass                75 : 0.0174 / 90 : 0.05858 / 95 : 0.10412 / 99 : 0.23868    
                                       radius             75 : 0.00477 / 90 : 0.00826 / 95 : 0.01121 / 99 : 0.01938    

                                                          time  
model physical_model filter            output_parameter         
KNN   MIST           scale_standard_14 mass              7.222  
                                       radius            7.222

In [6]:
compare_metrics(path_fine_tuned, value_rounding=3, relative_values=False,
                output_parameters=["mass", "radius"],
                physical_models=["MIST"],
                model_names=["KNN"],
                data_filters=["scale_standard_14"],
                categories=["Global"])

Results for the Global category with a value of 1.0


value range  RVE  \
model physical_model filter            output_parameter                        
KNN   MIST           scale_standard_14 mass                0.1 - 13.999  1.0   
                                       radius            -0.997 - 3.082  1.0   

                                                          RMSE    MAE  MedAE  \
model physical_model filter            output_parameter                        
KNN   MIST           scale_standard_14 mass              0.047  0.017  0.003   
                                       radius            0.005  0.003  0.002   

                                                         CORR  MAX_ER  \
model physical_model filter            output_parameter                 
KNN   MIST           scale_standard_14 mass               1.0   2.566   
                                       radius             1.0   0.126   

                                                                                                         Percentiles  \
model physical_model filter            output_parameter                                                                
KNN   MIST           scale_standard_14 mass               75 : 0.01276 / 90 : 0.04228 / 95 : 0.07758 / 99 : 0.19843    
                                       radius              75 : 0.0037 / 90 : 0.00691 / 95 : 0.00988 / 99 : 0.01799    

                                                          time  
model physical_model filter            output_parameter         
KNN   MIST           scale_standard_14 mass              9.006  
                                       radius            9.006

Slight improvement after fine-tuning

### Random forest

In [11]:
compare_metrics(path_default, value_rounding=3, relative_values=False,
                output_parameters=["mass", "radius"],
                physical_models=["MIST"],
                model_names=["random_forest"],
                data_filters=["scale_standard_14"],
                categories=["Global"])

Results for the Global category with a value of 1.0


value range  \
model         physical_model filter            output_parameter                   
random_forest MIST           scale_standard_14 mass                0.1 - 13.999   
                                               radius            -0.997 - 3.082   

                                                                 RVE   RMSE  \
model         physical_model filter            output_parameter               
random_forest MIST           scale_standard_14 mass              1.0  0.033   
                                               radius            1.0  0.005   

                                                                   MAE  MedAE  \
model         physical_model filter            output_parameter                 
random_forest MIST           scale_standard_14 mass              0.011  0.003   
                                               radius            0.002  0.001   

                                                                 CORR  MAX_ER  \
model         physical_model filter            output_parameter                 
random_forest MIST           scale_standard_14 mass               1.0   2.322   
                                               radius             1.0   0.130   

                                                                                                                 Percentiles  \
model         physical_model filter            output_parameter                                                                
random_forest MIST           scale_standard_14 mass               75 : 0.00984 / 90 : 0.02525 / 95 : 0.04683 / 99 : 0.13986    
                                               radius              75 : 0.00291 / 90 : 0.00576 / 95 : 0.00839 / 99 : 0.0168    

                                                                    time  
model         physical_model filter            output_parameter           
random_forest MIST           scale_standard_14 mass              288.308  
                                               radius            288.308

In [12]:
compare_metrics(path_fine_tuned, value_rounding=3, relative_values=False,
                output_parameters=["mass", "radius"],
                physical_models=["MIST"],
                model_names=["random_forest"],
                data_filters=["scale_standard_14"],
                categories=["Global"])

Results for the Global category with a value of 1.0


value range  \
model         physical_model filter            output_parameter                   
random_forest MIST           scale_standard_14 mass                0.1 - 13.999   
                                               radius            -0.997 - 3.082   

                                                                 RVE   RMSE  \
model         physical_model filter            output_parameter               
random_forest MIST           scale_standard_14 mass              1.0  0.032   
                                               radius            1.0  0.005   

                                                                   MAE  MedAE  \
model         physical_model filter            output_parameter                 
random_forest MIST           scale_standard_14 mass              0.011  0.003   
                                               radius            0.002  0.001   

                                                                 CORR  MAX_ER  \
model         physical_model filter            output_parameter                 
random_forest MIST           scale_standard_14 mass               1.0   1.868   
                                               radius             1.0   0.325   

                                                                                                                 Percentiles  \
model         physical_model filter            output_parameter                                                                
random_forest MIST           scale_standard_14 mass                75 : 0.00949 / 90 : 0.02452 / 95 : 0.0448 / 99 : 0.13388    
                                               radius             75 : 0.00267 / 90 : 0.00581 / 95 : 0.00874 / 99 : 0.01753    

                                                                    time  
model         physical_model filter            output_parameter           
random_forest MIST           scale_standard_14 mass              902.746  
                                               radius            902.746

Better MAX_ER for mass but worse for radius.

## Comparison between fine tuned models

| Parameter      | mass_filtered_smaller_14                                           |
| -------------  | ------------------------------------------------------------------ |
| age            | Range : 5.0 - 10.3, Mean : 8.4693, Median : 8.5                    |
| mass           | star_mass  : Range : 0.1 - 13.9995, Mean : 3.3831, Median : 2.0289 |
| log_teff       | log_Teff  : Range : 3.34 - 5.2138, Mean : 3.9383, Median : 3.867   |
| log_g          | log_g  : Range : -1.08 - 7.9405, Mean : 3.5574, Median : 3.9537    |
| log_R          | log_R  : Range : -1.86 - 3.0828, Mean : 0.6007, Median : 0.5357    |

### Mass parameter

In [15]:
compare_metrics(path_fine_tuned, value_rounding=3, relative_values=False,
                output_parameters=["mass"],
                physical_models=["MIST"],
                categories=["Global"],
                model_names=["KNN", "random_forest"])

Results for the Global category with a value of 1.0


value range  \
model         physical_model filter            output_parameter                 
KNN           MIST           mass_smaller_14   mass              0.1 - 13.999   
                             scale_standard_14 mass              0.1 - 13.999   
random_forest MIST           mass_smaller_14   mass              0.1 - 13.999   
                             scale_standard_14 mass              0.1 - 13.999   

                                                                 RVE   RMSE  \
model         physical_model filter            output_parameter               
KNN           MIST           mass_smaller_14   mass              1.0  0.058   
                             scale_standard_14 mass              1.0  0.047   
random_forest MIST           mass_smaller_14   mass              1.0  0.031   
                             scale_standard_14 mass              1.0  0.032   

                                                                   MAE  MedAE  \
model         physical_model filter            output_parameter                 
KNN           MIST           mass_smaller_14   mass              0.020  0.005   
                             scale_standard_14 mass              0.017  0.003   
random_forest MIST           mass_smaller_14   mass              0.011  0.003   
                             scale_standard_14 mass              0.011  0.003   

                                                                 CORR  MAX_ER  \
model         physical_model filter            output_parameter                 
KNN           MIST           mass_smaller_14   mass               1.0   2.562   
                             scale_standard_14 mass               1.0   2.566   
random_forest MIST           mass_smaller_14   mass               1.0   1.852   
                             scale_standard_14 mass               1.0   1.868   

                                                                                                                 Percentiles  \
model         physical_model filter            output_parameter                                                                
KNN           MIST           mass_smaller_14   mass                 75 : 0.01728 / 90 : 0.0509 / 95 : 0.08791 / 99 : 0.2393    
                             scale_standard_14 mass               75 : 0.01276 / 90 : 0.04228 / 95 : 0.07758 / 99 : 0.19843    
random_forest MIST           mass_smaller_14   mass               75 : 0.00931 / 90 : 0.02399 / 95 : 0.04372 / 99 : 0.13158    
                             scale_standard_14 mass                75 : 0.00949 / 90 : 0.02452 / 95 : 0.0448 / 99 : 0.13388    

                                                                     time  
model         physical_model filter            output_parameter            
KNN           MIST           mass_smaller_14   mass                11.112  
                             scale_standard_14 mass                 9.006  
random_forest MIST           mass_smaller_14   mass              1986.565  
                             scale_standard_14 mass               902.746

### Radius parameter

In [16]:
compare_metrics(path_fine_tuned, value_rounding=3,
                output_parameters=["radius"],
                physical_models=["MIST"],
                categories=["Global"],
                model_names=["KNN", "random_forest"])

Results for the Global category with a value of 1.0


value range  \
model         physical_model filter            output_parameter                   
KNN           MIST           mass_smaller_14   radius            -0.997 - 3.082   
                             scale_standard_14 radius            -0.997 - 3.082   
random_forest MIST           mass_smaller_14   radius            -0.997 - 3.082   
                             scale_standard_14 radius            -0.997 - 3.082   

                                                                 RVE   RMSE  \
model         physical_model filter            output_parameter               
KNN           MIST           mass_smaller_14   radius            1.0  0.005   
                             scale_standard_14 radius            1.0  0.005   
random_forest MIST           mass_smaller_14   radius            1.0  0.004   
                             scale_standard_14 radius            1.0  0.005   

                                                                   MAE  MedAE  \
model         physical_model filter            output_parameter                 
KNN           MIST           mass_smaller_14   radius            0.003  0.002   
                             scale_standard_14 radius            0.003  0.002   
random_forest MIST           mass_smaller_14   radius            0.002  0.001   
                             scale_standard_14 radius            0.002  0.001   

                                                                 CORR  MAX_ER  \
model         physical_model filter            output_parameter                 
KNN           MIST           mass_smaller_14   radius             1.0   0.179   
                             scale_standard_14 radius             1.0   0.126   
random_forest MIST           mass_smaller_14   radius             1.0   0.117   
                             scale_standard_14 radius             1.0   0.325   

                                                                                                                 Percentiles  \
model         physical_model filter            output_parameter                                                                
KNN           MIST           mass_smaller_14   radius             75 : 0.00346 / 90 : 0.00661 / 95 : 0.00922 / 99 : 0.01696    
                             scale_standard_14 radius              75 : 0.0037 / 90 : 0.00691 / 95 : 0.00988 / 99 : 0.01799    
random_forest MIST           mass_smaller_14   radius             75 : 0.00227 / 90 : 0.00472 / 95 : 0.00715 / 99 : 0.01498    
                             scale_standard_14 radius             75 : 0.00267 / 90 : 0.00581 / 95 : 0.00874 / 99 : 0.01753    

                                                                     time  
model         physical_model filter            output_parameter            
KNN           MIST           mass_smaller_14   radius              11.112  
                             scale_standard_14 radius               9.006  
random_forest MIST           mass_smaller_14   radius            1986.565  
                             scale_standard_14 radius             902.746

Random forest is the best model from all the fine tuned ones and even though the scaling gave better results with the default models, the results without scaling is slighlty better.